packages/cli/docs-resolver/src/utils/getImageFilepathsToUpload.ts

```diff
@@ -1,7 +1,22 @@
 import { docsYml } from "@fern-api/configuration-loader";
-import { AbsoluteFilePath } from "@fern-api/fs-utils";
+import { AbsoluteFilePath, resolve } from "@fern-api/fs-utils";
+import { DocsWorkspace } from "@fern-api/workspace-loader";
 
-export function collectFilesFromDocsConfig(parsedDocsConfig: docsYml.ParsedDocsConfiguration): Set<AbsoluteFilePath> {
+function shouldProcessIconPath(iconPath: string): boolean {
+    return iconPath.startsWith(".") || iconPath.startsWith("/");
+}
+
+function addIconToFilepaths(iconPath: string, filepaths: Set<AbsoluteFilePath>, docsWorkspace: DocsWorkspace): void {
+    if (shouldProcessIconPath(iconPath)) {
+        const absoluteIconPath = resolve(docsWorkspace.absoluteFilePath, iconPath);
+        filepaths.add(absoluteIconPath);
+    }
+}
+
+export function collectFilesFromDocsConfig(
+    parsedDocsConfig: docsYml.ParsedDocsConfiguration,
+    docsWorkspace: DocsWorkspace
+): Set<AbsoluteFilePath> {
```

Comment:
nit: generally using objects (named args) with multi-parameter functions is better